test/e2e/node_feature_discovery_test.go

```diff
@@ -500,6 +500,17 @@ var _ = NFDDescribe(Label("nfd-master"), func() {
 					}
 					eventuallyNonControlPlaneNodes(ctx, f.ClientSet).Should(MatchLabels(expectedLabels, nodes))
 
+					By("Verifying the CRD status data")
+
+					nf, err := nfdClient.NfdV1alpha1().NodeFeatures(f.Namespace.Name).Get(ctx, targetNodeName, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+
+					labelsMatch := len(expectedLabels[targetNodeName]) == nf.Status.NumberOfLabels
```

Comment:
This is simply checking a number of labels, but not the `NodeFeatureStatus` , please re-write to have a expected `NodeFeatureStatus` and check that the same is being returned